relayer/chains/cosmos/types.go

```diff
@@ -1,74 +1,28 @@
 package cosmos
 
 import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
 	clienttypes "github.com/cosmos/ibc-go/v3/modules/core/02-client/types"
-	chantypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
-	"github.com/cosmos/relayer/v2/relayer/processor"
 	"github.com/cosmos/relayer/v2/relayer/provider"
+	"go.uber.org/zap"
 )
 
 // ibcMessage is the type used for parsing all possible properties of IBC messages
 type ibcMessage struct {
-	messageType string
-	messageInfo interface{}
+	action string
+	info   ibcMessageInfo
 }
 
-// channelInfo contains useful channel information during channel state changes
-type channelInfo struct {
-	portID                string
-	channelID             string
-	counterpartyPortID    string
-	counterpartyChannelID string
-	connectionID          string
+type ibcMessageInfo interface {
+	parseAttrs(log *zap.Logger, attrs []sdk.Attribute)
 }
 
-// channelKey returns the processor.ChannelKey from channelInfo
-func (c channelInfo) channelKey() processor.ChannelKey {
-	return processor.ChannelKey{
-		ChannelID:             c.channelID,
-		PortID:                c.portID,
-		CounterpartyChannelID: c.counterpartyChannelID,
-		CounterpartyPortID:    c.counterpartyPortID,
-	}
-}
-
-type connectionInfo struct {
-	connectionID             string
-	clientID                 string
-	counterpartyClientID     string
-	counterpartyConnectionID string
-}
-
-func (c connectionInfo) connectionKey() processor.ConnectionKey {
-	return processor.ConnectionKey{
-		ConnectionID:             c.connectionID,
-		ClientID:                 c.clientID,
-		CounterpartyConnectionID: c.counterpartyConnectionID,
-		CounterpartyClientID:     c.counterpartyClientID,
-	}
-}
+// alias types to the provider types, used for adding parser methods
+type packetInfo provider.PacketInfo
 
-// packetInfo contains pertinent packet information for constructing IBC messages for the counterparty.
-type packetInfo struct {
-	// Packet is the IBC conformant Packet.
-	packet chantypes.Packet
+type channelInfo provider.ChannelInfo
 
-	// if message is a MsgRecvPacket, this is needed to construct MsgAcknowledgement for counterparty.
-	ack []byte
-
-	channelOrdering string
-	connectionID    string
-}
-
-// channelKey returns the processor.ChannelKey from packetInfo
-func (p packetInfo) channelKey() processor.ChannelKey {
-	return processor.ChannelKey{
-		ChannelID:             p.packet.SourceChannel,
-		PortID:                p.packet.SourcePort,
-		CounterpartyChannelID: p.packet.DestinationChannel,
-		CounterpartyPortID:    p.packet.DestinationPort,
-	}
-}
+type connectionInfo provider.ConnectionInfo
```

Comment:
It's awkward to have a type declared far away from its method definitions; and I would further argue that `types.go` is an anti-pattern (presumably cargo-culted from other packages we depend on) that we should discontinue, on the basis of resemblance to a [utils package](https://dave.cheney.net/2019/01/08/avoid-package-names-like-base-util-or-common).

The `...Info` types could move into `event_parser.go` near the method definitions, or perhaps these types and their methods could move into a new `info.go` file.

Reply:
done